themes/default/content/blog/resource-methods-for-pulumi-packages/index.md

```diff
@@ -0,0 +1,609 @@
+---
+title: "Introducing Resource Methods for Pulumi Packages"
+date: 2021-10-11T08:00:00-07:00
+draft: true
+meta_desc: It's now possible to provide resource methods from Pulumi Packages
+meta_image: pulumi-mlc-packages.png
+authors:
+    - justin-vanpatten
+tags:
+    - packages
+---
+
+It's now possible to provide resource methods from Pulumi Packages. Resource methods are similar to functions, but instead of being exposed as top-level functions in a module, methods are exposed as methods on a resource class. This allows for a more object oriented approach to exposing functionality -- operations performed by a resource (that potentially use the resource's state) can now be exposed as methods on the resource. Resource methods can be implemented once, in your language of choice, and made available to users in all Pulumi languages.
```

Comment:
```suggestion
It's now possible to provide resource methods from Pulumi Packages. Resource methods are similar to functions, but instead of being exposed as top-level functions in a module, methods are exposed as methods on a resource class. This allows for a more object-oriented approach to exposing functionality&mdash;operations performed by a resource (that potentially use the resource's state) can now be exposed as methods on the resource. Resource methods can be implemented once, in your language of choice, and made available to users in all Pulumi languages.
```